core/chains/evm/gas/models.go

```diff
@@ -186,8 +186,13 @@ func (b Block) MarshalJSON() ([]byte, error) {
 	})
 }
 
+var ErrMissingBlock = errors.New("missing block")
+
 // UnmarshalJSON unmarshals to a Block
 func (b *Block) UnmarshalJSON(data []byte) error {
+	if string(data) == "null" {
```

Comment:
I thought we were hitting line 200 with `Number: null`: `hexutil.DecodeBig(bi.Number)`
Did I misread?

Reply:
The error in question from the logs is definitely from 200/202:
> "err":"failed to decode block number while unmarshalling block, got: 'null': empty hex string","errVerbose":"empty hex string\nfailed to decode block number while unmarshalling block,

Reply:
Yes, geth returns 200 with an empty block

Reply:
I'm concerned about embedding high level assumptions of client behavior down inside of low level JSON code, and hand coding JSON parsing seems brittle and unnecessary anyways. What about just using a pointer instead of a struct value, so that the json lib handles it for us?
```go
var bi *blockInternal
if err := json.Unmarshal(data, &bi); err != nil {
	return errors.Wrapf(err, "failed to unmarshal to blockInternal, got: '%s'", data)
}
if bi == nil {
	return errors.WithStack(ErrMissingBlock)
}
```